bhv/symbolic.py

```diff
@@ -617,6 +617,44 @@ def expected_active_fraction(self, **kwargs):
         return 1. - PoiBin([v.expected_active_fraction(**kwargs) for v in self.vs]).cdf(len(self.vs)//2)
 
 
+@dataclass
+class Representative(SymbolicBHV):
+    vs: list[SymbolicBHV]
+
+    def labeled_children(self, **kwargs):
+        return list(zip(self.vs, map(str, range(len(self.vs)))))
+
+    def reconstruct(self, *cs):
+        return Representative(list(cs))
+
+    def show(self, **kwargs):
+        args = format_list((v.show(**kwargs) for v in self.vs), **{k: kwargs[k] for k in ["indent", "aindent", "newline_threshold"] if k in kwargs})
+        return kwargs.get("impl", "") + f"representative({args})"
+
+    def instantiate(self, **kwargs):
+        return kwargs.get("bhv").representative([v.execute(**kwargs) for v in self.vs])
+
+    # def expected_active_fraction(self, **kwargs):
+    #     from .poibin import PoiBin
+    #    return 1. - PoiBin([v.expected_active_fraction(**kwargs) for v in self.vs]).cdf(len(self.vs)//2)
+
+    def expected_error(self, x: Var):
+        """
+        Gives expected bit error rate between an expression that contains only representative operators and random
+        hypervectors, and a given random hypervector.
+        """
+        return self._expected_error(self, x)
+
+    def _expected_error(self, r, x):
```

Comment:
I need help finding the signature here. It's (SymbolicBHV, SymbolicBHV, SymbolicBHV) -> numeric or something now?

Reply:
Possibly, you want to use `self` instead of `r`. In other parts of the symbolic library, I use Rand's id or Var's str instead of the SymbolicBHV type, but maybe that needs to be re-evaluated.

Reply:
The signature should be ```def _expected_error(self, r: SymbolicBHV, x:Var) -> numeric```.  More specific even ```def _expected_error(self, r:Union[Representative, Var], x:Var) -> numeric```.  X should be the hypervector you want to compare against. I'll add the signature. 

I can change it to use self here. But this requires to also add an expected error method to the Var class.

Reply:
Where is `self` used? Maybe this can become a static or class method. Or maybe `r` can be `self`.

Do you think the expected error can be implemented for (all?) other functions? If so, we may no longer need the boolean hypervectors, and we could just use the symbolic methods!

Reply:
Currently I added the expected_error to the 'Var' class. Check out whether you like it. 

If not we can make the previous ```_expected_error``` method a static method indeed. I'm not sure whether it can be implemented for all other functions. I think that would bring us back to problems we encountered with the normal form, such as difficulties with Maj(a, a, b).

Reply:
We'll need similar machinery here and for the functions on a "Majority Tree".
I'm not sure what the best path forward is. Does representative flattens into weighted representative?